### ISSUE TITLE
Support Apple M1 architecture

### ISSUE THREAD

**Issue by programmarchy** [justjavac/dvm#51]
Deno releases now include aarch64 builds, so support for Apple M1 processors is within reach.

Currently, `dvm` can be used to install x86_64 builds of Deno which seem to run OK in Rosetta. However, it'd be ideal if it installed aarch64 builds instead.

I was able to compile `dvm` on my M1 machine and install it with a [modified script](https://gist.github.com/programmarchy/cc071c9fd7ee88e6ab6097956daa6b3d). However, to distribute `dvm` for Apple M1, the GitHub CI script would need to be modified as well. It may be possible for the rust toolchain to cross-compile for that target (still looking into it).

To recap, steps needed to support Apple M1:

- [ ] modify dvm src to download aarch64 binaries (`#[cfg(target_arch = "aarch64")]`)
- [ ] modify GitHub Actions script to cross-compile a target for aarch64
- [ ] modify install script to install the aarch64 target when detected

Perhaps `dvm` could also accept an `--arch` flag to allow users to install either aarch64 or x86_64 binaries.

**Comment by code-ape** [comment]
Any news on this? Recently came across an issue seemingly due to lack of ARM64 binary support. Thanks!

**Comment by justjavac** [comment]
The codebase already supports ARM64, but GitHub actions do not support ARM64 environments. AND I don't have a M1 mac. 😂

**Comment by code-ape** [comment]
@justjavac could you just use the Apple ARM64 builds that publicly Deno publishes? They're on the Github repo and have been built and added to releases since v1.6.0.

Example: Asset `deno-aarch64-apple-darwin.zip` from [Deno release v1.12.3](https://github.com/denoland/deno/releases/tag/v1.21.3).

**Comment by programmarchy** [comment]
@code-ape The linked PR does exactly that. This repo doesn’t provide arm64 builds *of dvm*, but you can either build it yourself or install via cargo (see the readme).

**Comment by code-ape** [comment]
@programmarchy I see what you did now! You have it to where if you have an aarch64 build of dvm then it pulls an aarch64 build of Deno. What I was hoping for was that dvm would detect the aarch64 architecture even if it was x86_64 and then download the proper one.

An easy way to do this would be to have run the shell command `uname -m` and check if it matches `arm64` and if so download the `deno-aarch64-apple-darwin.zip` instead of `deno-x86_64-apple-darwin.zip`. You'd just need to update the contents of `src/commands/install.rs` to have a function `fn get_archive_name()` instead of `const ARCHIVE_NAME` which would return a fixed string for Windows and Linux but for MacOS to run the shell command, check the output, and then either return the x86_64 or aarch64 string.
